src/Aspire.Hosting.Analyzers/Aspire.Hosting.Analyzers.csproj

```diff
@@ -0,0 +1,39 @@
+<Project Sdk="Microsoft.NET.Sdk">
+
+  <PropertyGroup>
+    <TargetFramework>netstandard2.0</TargetFramework>
+    <IncludeBuildOutput>false</IncludeBuildOutput>
+    <IsRoslynComponent>true</IsRoslynComponent>
+    <EnforceExtendedAnalyzerRules>true</EnforceExtendedAnalyzerRules>
+    <UsePublicApiAnalyzers>false</UsePublicApiAnalyzers>
+    <IsPackable>false</IsPackable>
+    <IsShipping>false</IsShipping>
+    <LangVersion>12.0</LangVersion>
+    <!--
+      Analyzer projects don't need XML docs
+      CS1573: Parameter 'parameter' has no matching param tag in the XML comment for 'parameter' (but other parameters do)
+      CS1591: Missing XML comment for publicly visible type or member 'Type_or_Member'
+      CS1712: Type parameter 'type_parameter' has no matching typeparam tag in the XML comment on 'type_or_member' (but other type parameters do)
+    -->
+    <NoWarn>$(NoWarn),1573,1591,1712</NoWarn>
+    <!--
+      We manually control the dependency versions that the analyzer uses as it needs to be loaded into the compiler host, e.g. Visual Studio,
+      which may have a different set of dependencies.
+    -->
+    <ManagePackageVersionsCentrally>false</ManagePackageVersionsCentrally>
+  </PropertyGroup>
+
+  <ItemGroup>
+    <Compile Include="..\Aspire.Hosting\ApplicationModel\ModelName.cs" Link="ModelName.cs" />
+    <Compile Include="..\Shared\StringComparers.cs" Link="StringComparers.cs" />
+  </ItemGroup>
+
+  <ItemGroup>
+     <PackageReference Include="Microsoft.CodeAnalysis.CSharp" Version="4.11.0" PrivateAssets="All" /> 
```

Comment:
Is 4.11 too new? It was only shipped a month ago.

If someone is using net8 and an older VS, this might break them.

Reply:
How do we determine what the correct version to use is?

Reply:
ASP.NET Core seems to be using 3.3.1 because it says some analyzers need to be used in VS2019, but perhaps that's out of date? https://github.com/dotnet/aspnetcore/blob/e25539b691b0425f94f66818ea0a0b4820f44804/eng/Versions.props#L247-L253

Reply:
@jaredpar @dsplaisted what's the right way to determine which version of `Microsoft.CodeAnalysis.CSharp` our analyzer project should reference?

Reply:
I think you take the version that aligns with the minimum VS or .NET SDK dependency.

https://learn.microsoft.com/dotnet/aspire/fundamentals/setup-tooling says VS 17.10 is the minimum VS. That aligns with Roslyn 4.10. And .NET 8 SDK, I assume 8.0.100, which aligns with Roslyn 4.8.

So a good starting point would be `4.8.0`, I'd think. I don't imagine us needing to go back further since we only support .NET 8+.

Reply:
Do your analyzers ship in a package or as part of the .NET SDK? If it's a separate package then @eerhardt is correct. If it's part of the .NET SDK then you can start taking a dependency on latest.

Reply:
A package, thanks.